### PR TITLE
[scaffolder] Make the `scaffolder/next` work the default

### LINKED ISSUE #18251: Make `scaffolder/next` the default
Time has come.

It's been in docs for a while, and I think we've had enough feedback now and I think we're ready to make the switch to the new scaffolder work in mainline release 1.16.

It would be good to get this work shipped in the earliest possible `-next` release of `1.16`, so let's aim to merge this after the mainline release of 1.15 which is Tuesday 20th.

### PR DESCRIPTION
**Pull request by benjdlambert**
Fixes #18251 

This also moves all the old stuff under `/alpha` imports as `Legacy*` so that people don't have to migrate straight away. But the intention is that this will be removed the mainline release after as a minor change.

*todo*
- Work out why the tests are throwing typescript errors for `RepoUrlPicker`
- Remove the duplicate code that's appeared in this PR as we're keeping that in @public
- Create changesets for all the changes
- Update docs

### CODE REVIEW

File: plugins/scaffolder/src/components/TemplateEditorPage/DryRunResults/DryRunResultsView.tsx [hunk 1]
```diff
@@ -24,11 +24,13 @@ import Tab from '@material-ui/core/Tab';
...
-import { TaskStatusStepper } from '../../TaskPage/TaskPage';
-import { TaskPageLinks } from '../../TaskPage/TaskPageLinks';
+
 import { useDryRun } from '../DryRunContext';
-import { FileBrowser } from '../../FileBrowser';
+
 import { DryRunResultsSplitView } from './DryRunResultsSplitView';
+import { FileBrowser } from '../../../components/FileBrowser';
+import { TaskStatusStepper } from '../../../legacy/TaskPage/TaskPage';
+import { TaskPageLinks } from '../../../legacy/TaskPage/TaskPageLinks';
```
Comment (Rugvip) [inline plugins/scaffolder/src/components/TemplateEditorPage/DryRunResults/DryRunResultsView.tsx:33]: Would be good to flip this around right? trying to get to a state where we can simply delete the  `legacy` folder once it's ready.

File: packages/app/src/App.tsx [hunk 2]
```diff
@@ -219,11 +219,11 @@ const routes = (
         <LightBox />
       </TechDocsAddons>
     </Route>
-    <FeatureFlagged with="scaffolder-next-preview">
+    <FeatureFlagged with="scaffolder-legacy">
       <Route
         path="/create"
         element={
-          <NextScaffolderPage
+          <LegacyScaffolderPage
```
Comment (Rugvip) [inline packages/app/src/App.tsx:226]: Is the `NextScaffolderPage` a drop-in replacement for the legacy one? as in will we roll this out as a breaking change or not?

**Review by Rugvip — COMMENTED** [review]
Some early stuff, gief API reports 😁

File: packages/app/src/App.tsx (hunk 2 above)
Comment (benjdlambert) [reply packages/app/src/App.tsx:226]: Unfortunately the breaking change is not in the types that we own, it's in the FieldExtension types. So yes it's a breaking change, and yes it's a drop in here, but not everywhere.

Comment (benjdlambert) [reply packages/app/src/App.tsx:226]: Unfortunately the breaking change is not in the types that we own, it's in the FieldExtension types. So yes it's a breaking change, and yes it's a drop in here, but not everywhere.

File: plugins/scaffolder/src/components/TemplateEditorPage/DryRunResults/DryRunResultsView.tsx (hunk 1 above)
Comment (benjdlambert) [reply plugins/scaffolder/src/components/TemplateEditorPage/DryRunResults/DryRunResultsView.tsx:33]: yeah indeed. that was one of my thoughts.

**Comment by benjdlambert** [comment]
update on this is we're probably going to look at shipping this without a breaking change for consumers, and this will be re-opened in due corse.

**Pull request closed without merging**